internal/releaseinfo/releaseinfo.go

```diff
@@ -12,8 +15,20 @@ type ReleaseInfo struct {
 func FetchLatestRelease() (*ReleaseInfo, error) {
 	url := "https://glasskube.dev/release.json"
 
-	resp, _ := http.Get(url)
-
+	resp, err := http.Get(url)
+	if err != nil {
+		if err := httperror.CheckResponse(resp); err != nil {
+			switch {
+			case httperror.IsNetworkError(err):
+				return nil, fmt.Errorf("Network Timeout, check your network %v: %v", url, err)
+			case httperror.Is(err, http.StatusServiceUnavailable):
+				return nil, fmt.Errorf("Could not connect to the glasskube server. Please check your internet connection and try again. %v: %v", url, err)
+			default:
+				return nil, fmt.Errorf("Failed to fetch glasskube version %v: %v", url, err)
+			}
+		}
+		return nil, err
+	}
```

Comment:
Unfortunately, I think there are some mistakes here. This is understandable because network errors in Go can be a little confusing…

1. If `err` is initially not `nil`, this means that the HTTP request has failed and the response object should not be used (it is `nil`). In this case this leads to a SIGSEGV because `httperror.CheckResponse` requires that the response is not `nil`.
2. `httperror.CheckResponse` does not return a `net.Error`, so calling `httperror.IsNetworkError` on the error returned by it can never be true.

I can think of two ways to improve this: 
1. Locally, by checking `err` first and then handling a possible HTTP error
```go
if err != nil {
  /* check network error here */
} else if err := httperror.CheckResponse(resp); err != nil {
  /* handle HTTP error here */
}
```
2. Globally, by changing `httperror.CheckResponse` to accept and return both an error and a response: 
```go
func CheckResponse(response *http.Response, err error) (*http.Response, error) {
	if err != nil {
		return response, err
	} else if IsErrorResponse(response) {
		return response, &statusError{response.Status, response.StatusCode}
	} else {
		return response, nil
	}
}
```

This is very neat because it can be used like this:

```go
resp, err := httperror.CheckResponse(http.Get(url))
```

I think option (2) would be preferable, but if you rather want to do option (1) that's fine by me too.